packages/mux-player/src/index.ts

```diff
@@ -0,0 +1,453 @@
+  SM: "small",
+};
+
+function getPlayerSize(el: Element) {
+  const muxPlayerRect = el.getBoundingClientRect();
+  return muxPlayerRect.width < SMALL_BREAKPOINT
+    ? MediaChromeSizes.SM
+    : MediaChromeSizes.LG;
+}
+
+const MuxVideoAttributes = {
+  ENV_KEY: "env-key",
+  DEBUG: "debug",
+  PLAYBACK_ID: "playback-id",
+  METADATA_URL: "metadata-url",
+  PREFER_MSE: "prefer-mse",
+  METADATA_VIDEO_ID: "metadata-video-id",
+  METADATA_VIDEO_TITLE: "metadata-video-title",
+  METADATA_VIEWER_USER_ID: "metadata-viewer-user-id",
+  BEACON_DOMAIN: "beacon-domain",
+  TYPE: "type",
+  STREAM_TYPE: "stream-type",
+  START_TIME: "start-time",
+};
+
+const PlayerAttributes = {
+  DEFAULT_SHOW_CAPTIONS: "default-show-captions",
+  PRIMARY_COLOR: "primary-color",
+  SECONDARY_COLOR: "secondary-color",
+  FORWARD_SEEK_OFFSET: "forward-seek-offset",
+  BACKWARD_SEEK_OFFSET: "backward-seek-offset",
+};
+
+function getProps(el: MuxPlayerElement, props?: any): MuxTemplateProps {
+  return {
+    debug: el.debug,
+    envKey: el.envKey,
+    playbackId: el.playbackId,
+    poster: el.poster,
+    startTime: el.startTime,
+    streamType: el.streamType,
+    primaryColor: el.primaryColor,
+    secondaryColor: el.secondaryColor,
+    forwardSeekOffset: el.forwardSeekOffset,
+    backwardSeekOffset: el.backwardSeekOffset,
+    defaultShowCaptions: el.defaultShowCaptions,
+    playerSize: getPlayerSize(el),
+    hasCaptions: !!getCcSubTracks(el).length,
+    showLoading: showLoading(el),
+    supportsAirPlay: false,
+    supportsVolume: false,
+    ...props,
+  };
+}
+
+const MuxVideoAttributeNames = Object.values(MuxVideoAttributes);
+const PlayerAttributeNames = Object.values(PlayerAttributes);
+const playerSoftwareVersion = getPlayerVersion();
+const playerSoftwareName = "mux-player";
+
+// Until real private properties are supported create private internals.
+const internals = new WeakMap();
+
+class MuxPlayerElement extends VideoApiElement {
+  static get observedAttributes() {
+    return [
+      ...(VideoApiElement.observedAttributes ?? []),
+      ...MuxVideoAttributeNames,
+      ...PlayerAttributeNames,
+    ];
+  }
+
+  constructor() {
+    super();
+    internals.set(this, new MuxPlayerInternal(this));
+  }
+
+  connectedCallback() {
+    internals.get(this).connectedCallback();
+  }
+
+  disconnectedCallback() {
+    internals.get(this).disconnectedCallback();
+  }
+
+  attributeChangedCallback(
+    attrName: string,
+    oldValue: string | null,
+    newValue: string
+  ) {
+    super.attributeChangedCallback(attrName, oldValue, newValue);
+
+    if (MuxVideoAttributeNames.includes(attrName)) {
+      this.video?.setAttribute(attrName, newValue);
+    }
+  }
+
+  get primaryColor() {
+    return this.getAttribute(PlayerAttributes.PRIMARY_COLOR);
+  }
+
+  get secondaryColor() {
+    return this.getAttribute(PlayerAttributes.SECONDARY_COLOR);
+  }
+
+  get forwardSeekOffset() {
+    return this.getAttribute(PlayerAttributes.FORWARD_SEEK_OFFSET);
```

Comment:
we need defaults here for `forwardSeekOffset` and `backwardSeekOffset`. I **_think_** adding them here should get picked up everywhere else.